include/aws/crt/ImdsClient.h

```diff
@@ -0,0 +1,341 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/crt/DateTime.h>
+#include <aws/crt/Exports.h>
+#include <aws/crt/Types.h>
+#include <functional>
+
+struct aws_array_list;
```

Comment:
Should these be publicly named here? Couldn't we just add struct in front of their declarations in the C++ types?